### ISSUE TITLE
Memory unmmaped when Process instance out of scope

### ISSUE THREAD

**Issue by dummy0stud** [DarthTon/Blackbone#63]
Hello,
```
{
  blackbone::Process* proc = new blackbone::Process();
  remotePE = proc->memory().Allocate(pe_size);
  status = proc->memory().Write(remotePE.ptr(), pe_size, pe_addr);
}
```
By calling this line we should be able to create a pointer that is not destroyed when the scope is destroyed, but this is not happening, when my_function returns the memory allocated inside remote process is unmapped, is Detach() being called implicitly? How can I keep memory mmaped inside remote process? You have any plans to release remote inline hooking anytime soon?

**Comment by DarthTon** [comment]
```
remotePE.Release();
```
Remote inline hooking most likely won't be released any time soon.

**Comment by dummy0stud** [comment]
Release() worked, thanks, but I'm curious, why even creating a pointer to the object the destructor is being called implicitly?

**Comment by DarthTon** [comment]
RAII stuff.

**Comment by dummy0stud** [comment]
There is any way to use AsmHelper64 using x86 of blackbone.lib? when I try to #define USE64 compiler gives me the following error

```
2>  TestApp.cpp(48): note: due to following members:
2>  TestApp.cpp(48): note: 'void blackbone::AsmHelperBase::ExitThreadWithStatus(size_t,size_t)': is abstract
```

**Comment by DarthTon** [comment]
1. Update to recent c4ea198479f8959238fcf1180ce7bdd751e5e107 commit
2. Include AsmHelper64.cpp into build under win32 solution platfrom.
3. Declare AsmHelper64 explicitly.

**Comment by dummy0stud** [comment]
thanks!

**Comment by DarthTon** [comment]
Forgot one file. Fixed in 08d6a0eb35c7bfc5bb1281be26e57a7e782a1547
